tracetools_trace/tracetools_trace/tools/lttng_impl.py

```diff
@@ -342,12 +342,14 @@ def setup(
             # Per-user buffer
             buffer_type=lttngpy.LTTNG_BUFFER_PER_UID,
             channel_name=channel_name,
-            # Discard, do not overwrite
-            overwrite=0,
-            # We use 2 sub-buffers because the number of sub-buffers is pointless in discard mode,
-            # and switching between sub-buffers introduces noticeable CPU overhead
+            # Overwrite if snapshot mode, otherwise discard
+            overwrite=snapshot_mode,
```

Comment:
I would just make this an explicit conversion from `bool` to `int`:

```suggestion
            overwrite=int(snapshot_mode),
```